src/map/lua/lua_baseentity.cpp

```diff
@@ -2452,6 +2453,47 @@ inline int32 CLuaBaseEntity::getAngle(lua_State *L)
     return 1;
 }
 
+/************************************************************************
+*  Function: getCardinalQuadrant()
+*  Purpose : Will return a index of what direction you are facing from a target.
+*  Example : if player:getCardinalQuadrant(target) >= 7 and player:getCardinalQuadrant(target) <=11 then
+*  Example :     print("can use sneak attack!!!")
+*  Example : end
+*  Notes   : index is in scripts/globals/status
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getCardinalQuadrant(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isuserdata(L, 1));
+
+    CLuaBaseEntity* PLuaBaseEntity = Lunar<CLuaBaseEntity>::check(L, 1);
+
+    auto angle = PLuaBaseEntity->GetBaseEntity()->loc.p.rotation - getangle(PLuaBaseEntity->GetBaseEntity()->loc.p, m_PBaseEntity->loc.p);
+    auto rotRad = rotationToRadian(angle);
+    auto cardinalAngle = radianToRotation(rotRad);
+    uint8 cardinalQuadrant = 0;
+
+    if ((cardinalAngle >= 0 && cardinalAngle < 8) || (cardinalAngle > 248 && cardinalAngle <= 256)){ cardinalQuadrant = 1; }// N
+    else if (cardinalAngle <= 248 && cardinalAngle > 216){ cardinalQuadrant = 2; } // NNE
+    else if (cardinalAngle <= 216 && cardinalAngle > 200){ cardinalQuadrant = 3; } // NE
+    else if (cardinalAngle <= 200 && cardinalAngle > 184){ cardinalQuadrant = 4; } // ENE
+    else if (cardinalAngle <= 184 && cardinalAngle > 168){ cardinalQuadrant = 5; } // E
+    else if (cardinalAngle <= 168 && cardinalAngle > 152){ cardinalQuadrant = 6; } // ESE
+    else if (cardinalAngle <= 152 && cardinalAngle > 136){ cardinalQuadrant = 7; } // SE
+    else if (cardinalAngle <= 136 && cardinalAngle > 120){ cardinalQuadrant = 8; } // SSE
+    else if (cardinalAngle <= 120 && cardinalAngle > 104){ cardinalQuadrant = 9; } // S
+    else if (cardinalAngle <= 104 && cardinalAngle > 88){ cardinalQuadrant = 10;}  // SSW
+    else if (cardinalAngle <= 88 && cardinalAngle > 72){ cardinalQuadrant = 11; }  // SW
+    else if (cardinalAngle <= 72 && cardinalAngle > 56){ cardinalQuadrant = 12; }  // WSW
+    else if (cardinalAngle <= 56 && cardinalAngle > 40){ cardinalQuadrant = 13; }  // W
+    else if (cardinalAngle <= 40 && cardinalAngle > 24){ cardinalQuadrant = 14; }  // WNW
+    else if (cardinalAngle <= 24 && cardinalAngle > 8){ cardinalQuadrant = 15; }   // NW
+    else if (cardinalAngle <= 8 && cardinalAngle > 0){ cardinalQuadrant = 16; }    // NNW
```

Comment:
There has to be a better way to do this.

Reply:
Yeah im sure someone smarter than me could think of a way, i tried, but it never seemed right